v2/handler.go

```diff
@@ -37,37 +37,61 @@ const (
 	Streaming
 )
 
-// HandlerSpec holds either a UnaryTransportHandler or StreamTransportHandler.
-type HandlerSpec struct {
+// TransportHandlerSpec holds either a UnaryTransportHandler or StreamTransportHandler.
+type TransportHandlerSpec struct {
 	t Type
 
 	unaryHandler  UnaryTransportHandler
 	streamHandler StreamTransportHandler
 }
 
+// EncodingHandlerSpec holds either UnaryEncodingHandler or StreamEncodingHandler
```

Comment:
Let’s avoid interleaving types and methods. With few exceptions, a type declaration should be followed by all of its methods. It might even be okay to rename this to transport.go and add encoding.go or similar with the mirror world types and methods.